app/main.py

```diff
@@ -3,78 +3,157 @@
 from pydantic import BaseModel
 from typing import Optional
 from random import randrange
-app = FastAPI()                         # get metoda
-                    
-my_posts = [{"title": "title of post 3", "content" : "Content of post 3", "id": 3},
-            {"title": "title of post 4", "content" : "Content of post   4", "id": 4}]
+import psycopg2
+from psycopg2.extras import RealDictCursor
+import time
 
+app = FastAPI()
 
+
+my_posts = [
+    {"title": "title of post 3", "content": "Content of post 3", "id": 3},
+    {"title": "title of post 4", "content": "Content of post   4", "id": 4},
+]
+
+
+# Funkcija za pretragu posta sa unetim id-om
 def find_post(id):
     for p in my_posts:
         if p["id"] == id:
             return p
 
+
+# Funkcija za pretragu indexa posta koji ima uneti id
 def find_index_post(id):
     for i, p in enumerate(my_posts):
-        if p['id'] == id:
+        if p["id"] == id:
             return i
 
+
+# Pravljenje objekta klase Post
 class Post(BaseModel):
     title: str
     content: str
     published: bool = True
-    rating: Optional[int] = None
-                    
-                                        # uvicorn main:app --reload Pravi live server i monitoruje promene u kodu
-@app.get("/")                           # Decorator
+
+
+# Pravljenje while petlje kako bi se usled neuspele konekcije sa bazon na svake 2 sekunde petlja ponovo izvrsi sve dok se ne ispravi problem.
+# Testirao sam sa menjanjem sifre i petlja se izvrasava sve dok nisam ukucao pravilnu sifru i sacuvao
+while True:
+
+    try:
+        conn = psycopg2.connect(
+            host="localhost",
+            database="fastapi",
+            user="postgres",
+            password="Ilija2002",
+            cursor_factory = RealDictCursor,
+        )
+        cursor = conn.cursor()
+        print("Database connection was succesfull!")
+        break
+    except Exception as e:
+        print("Database connection failed :(")
+        print("Error: ", e)
+        time.sleep(2)
```

Comment:
Usually, any kind of sleep() can cause unexpected behavior and always try to avoid using it. In this case, for demonstration purpose of retry mechanism to establish connection with db, it is fine, but we will see how to do the same in another approach